qr_generator/lib/domain/qr_generator/cubit/generator_cubit.dart

```diff
@@ -0,0 +1,54 @@
+import 'dart:ui';
+
+import 'package:bloc/bloc.dart';
+import 'package:dartz/dartz.dart';
+import 'package:equatable/equatable.dart';
+import 'package:qr_generator/domain/model/failure.dart';
+import 'package:qr_generator/domain/model/qr_code.dart';
+import 'package:qr_generator/domain/qr_generator/repository/qr_generator_repo.dart';
+
+part 'generator_state.dart';
+
+class GeneratorCubit extends Cubit<GeneratorState> {
+  GeneratorCubit(this.repository) : super(GeneratorInitial());
+
+  final QRGeneratorRepository repository;
+  late AppLifecycleState _lifecycleState;
+
+  void setLifeCycleEvent(AppLifecycleState lifecycleState) {
+    _lifecycleState = lifecycleState;
+  }
+
+  void generateQR() async {
+    if (_lifecycleState == AppLifecycleState.resumed) {
+      if (state is GeneratorError) {
+        emit(GeneratorInitial());
+      }
+      Either<Failure, QRCode> result = await repository.generateQRCode();
+
+      result.fold(
+        (Failure failure) => emit(
+          GeneratorError(errorMessage: failure.errorMessage),
+        ),
+        (QRCode qrCode) {
+          int secondsToExpire =
+              qrCode.expirationDate.difference(DateTime.now()).inSeconds;
+          _emitQRandExpiration(qrCode.data, secondsToExpire);
```

Comment:
suggestion: This secondsToExpire could technically be a negative number.  On line 46 we check to make sure it is > 1 and if not we emit a new generated qr.  I would be concerned what would happen if this was a negative number.  Would it not just loop forever?  This is unlikely to occur as far as I can tell but if it did, I think it would be bad.